### ISSUE TITLE
Windows 10 21H1 ISO file size difference

### ISSUE THREAD

**Issue by knownbyfew** [AveYo/MediaCreationTool.bat#53]
Hey,

So, I was looking to download the Windows 10 21H1 ISO and back when it was the latest version available the normal MCT generated ISO used to weigh up to about 4.29GB whereas the 21H1 ISO downloaded from MCT through your script is about 4.38GB, why is that?

Thanks,
Regards

**Comment by AveYo** [comment]
You most likely compared different builds. 
At the top of the script you can see the base build you're gonna get for different versions, and the changelog at the bottom of the readme lists any build changes.

**Comment by knownbyfew** [comment]
> At the top of the script you can see the base build you're gonna get for different versions, and the changelog at the bottom of the readme lists any build changes.

So do the builds downloaded through your script come with the latest updates installed or something?

**Comment by AveYo** [comment]
This script has often offered bleeding edge (official) builds long before new version release to the public, or after support has ended.  
Even now, 1703, 19H2, 20H1, 20H2, 21H1 have more recent build esd links (from WSUS), than the last supported MCT and public products.xml/cab for it did.  
That's what the "gibberish" at the end of the script is for: a highly condensed links update format I came up with.

**Comment by knownbyfew** [comment]
Ah, I see. Thanks for the response. 

This maybe off-topic, but I wanted to know - can automatic driver updates be disabled on Windows 10 Home? If so, how?

Thanks,
Regards

**Comment by AveYo** [comment]
@knownbyfew, sure.

The official registry-way that may or may not work:  `disable_drivers_update_via_wu.reg`
``` reg
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\Device Metadata]
"PreventDeviceMetadataFromNetwork"=dword:00000001

[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\DriverSearching]
"SearchOrderConfig"=dword:00000000

[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\WindowsUpdate]
"ExcludeWUDriversInQualityUpdate"=dword:00000001
``` 

The uncommon script by me:  `windows_driver_update_toggle.bat`
https://pastebin.com/cK8y4YEX

**Comment by knownbyfew** [comment]
Does the script work on Windows 10 Home? Have you tried it out on that personally? And also how am I supposed to create it? Do I just need to copy the contents from the pastebin, paste it in a notepad file and then save the script as .bat?

**Comment by AveYo** [comment]
read line no.2 .. can simply copy-paste into powershell and enter to run directly    
or just use download button on pastebin (top right) to prevent copy-paste errors  
I obviously tested it at that time when I updated it, and before I linked it here, but if for some reason it stops working, you can apply the registry keys above as well

**Comment by knownbyfew** [comment]
How different is the batch file from the registry keys? Is it the same but automated or is it a whole different method?

**Comment by AveYo** [comment]
different method. because microsoft has often ignored their own registry keys to control windows behavior, specially on consumer editions

**Comment by knownbyfew** [comment]
Hey, I was just curious, does your driver blocking script use options that readily available in Windows 10 Pro such as group policy editor or does it use options that are available to all editions of Windows to achieve this?

**Comment by AveYo** [comment]
common..  
it literally changes a value in the registry - that's all  
the rest is just for unprotecting the registry key - you can do that manually in regedit, but that's not convenient and you can mess up, hence the script

**Comment by knownbyfew** [comment]
Oh, so running the command once will inject the key into the registry and protect it and if you run the script again it will unprotect the key and delete it? Is that how it works?

**Comment by AveYo** [comment]
wouldn't it be better for you to refrain from doing stuff like this until you grasp some basics via some courses or something?   
most people will not have the time nor the patience to respond to questions that google could   
 
``` ps1
if ($null -ne $V -and 'Blocked' -eq $V.Prefixes[0]) {
   set-itemproperty $K 'Prefixes' 'd.' -type MultiString -force -ea 0
   write-host -fore 0xf -back 0xd "`n Driver Update [RESTORED] run again to block "
 } else {                                                                                                                                                                    
   set-itemproperty $K 'Prefixes' 'Blocked' -type MultiString -force -ea 0
   write-host -fore 0xf -back 0x2 "`n Driver Update [BLOCKED] run again to restore "
 }
```
=

``` ps1
if the value is 'Blocked' then 
   set it back to default 'd.' and write on-screen that it was RESTORED
else (if the value is not 'Blocked')   
    set it to 'Blocked', and write on-screen that it was BLOCKED  
```

**Comment by knownbyfew** [comment]
> wouldn't it be better for you to refrain from doing stuff like this until you grasp some basics via some courses or something? most people will not have the time nor the patience to respond to questions that google could

I apologize for the inconvenience.

**Comment by AveYo** [comment]
I said most people. That's not me ;)  
I do try to explain things. I'm not very good at ELI5, and for that I apologize instead.
But not being accustomed with scripts, registry & etc. means if something bad happens, you are gonna have a harder time to fix it, or even to describe the problem to others, so that's my blunt-_ish_ way of saying "do it at your own risk".

**Comment by knownbyfew** [comment]
Hey, sorry for the late response - I have actually started using Windows 10 Home with the registry and batch file that you had provided to me and it works great and does everything I want it to do. The only issue I face is that my "Check for updates" button in Settings seems to be broken, every time I search for a update, it downloads and installs it, but gets suck at "Installing 100%" but if I go back to the main menu and then come back to the updates page, the installation text goes blank, the only way to fix is to restart the settings app, only then can I see the "Check for updates" button again.

My question was - do you think it's an issue specific to my Windows 10 Home installation or is it because I used the registry file alongside the batch file as the registry file doesn't work the same in Windows 10 Home like it does on the Pro edition? Have you faced these issues personally? If not, do I reinstall a fresh copy of the latest version of Windows 10?

**Comment by AveYo** [comment]
It's to be expected that windows will not like any interference with it's shitty brainless automatic update. The smallest change can cause such issues you are experiencing, it's not because of your edition.  

If you can't live with the issue, revert the change.

**Comment by knownbyfew** [comment]
Nah, I'm good to be honest.

Never faced this issue back in Windows 10 with only using the registry file, so curiously struck. 

Next time I'll remember to choose Windows 10 Pro over Home for an easier time with useless automatic updates.
